docs: Update Pebble-formik ReadMe including npm package publishing (#26)
istreamlabs/pebble-formik: README.md

[![NPM](https://img.shields.io/npm/v/pebble-formik.svg)](https://www.npmjs.com/package/@istreamplanet/pebble-formik)

# Pebble-Formik

> [Pebble](https://pebble.istreamplanet.net/) form components wrapped in a [Formik](https://formik.org/docs/overview) Formik-friendly wrapper.

Reviewer: ```suggestion
[Pebble](https://pebble.istreamplanet.net/) form components wrapped in a [Formik](https://formik.org/docs/overview)-friendly wrapper.
```

npm install @istreamplanet/pebble-formik formik @istreamplanet/pebble node-sass react-router-dom --save
```

Include the Pebble foundation stylesheet in your layout template. (This is usually added at the outer most content wrapper of an app)

```shell
import '@istreamplanet/pebble/dist/Styles/foundation.scss';
```

Import the components into your application

```shell
import { FieldSelect, FormCard, FormLayoutRow } from '@istreamplanet/pebble-formik'`
```

Then add them in your React component as you would any other:

**The [FormCard](https://github.com/istreamlabs/pebble-formik/blob/main/src/FormCard.tsx) component is a great all-in-one form wrapper component. Please try it before building a Formik form from scratch.**

```shell
const MyComponent = () => (
  <Block>
    <FormCard initialValues={initialValues} onSubmit={handleSubmit} validate={validate}>
      <FormLayoutRow>
        <FieldSelect id="foo" name="foo" label="Foo" options={FooOptions} />
      </FormLayoutRow>
    </FormCard>
  </Block>
)
```

## Components

### [FormCard](https://github.com/istreamlabs/pebble-formik/blob/main/src/FormCard.tsx)

> FormCard was designed to be an all-in-one form rapper, including a prompt message if the user tries to navigate away from the form and a submit and reset button in the footer. (see component for all of the props)

Reviewer 2: Looks like I missed this the first pass, should be wrapper, not rapper.

Author: hahaha!!! oops

npm install @istreamplanet/pebble-formik formik @istreamplanet/pebble node-sass react-router-dom --save
```

Include the Pebble foundation stylesheet in your layout template. This is usually added at the outer most content wrapper of an app:

```shell
import '@istreamplanet/pebble/dist/Styles/foundation.scss';
```

Import the components into your application:

```shell
import { FieldSelect, FormCard, FormLayoutRow } from '@istreamplanet/pebble-formik';
```

Then add them in your React component as you would any other:

**The [FormCard](https://github.com/istreamlabs/pebble-formik/blob/main/src/FormCard.tsx) component is a great all-in-one form wrapper component. Please try it before building a Formik form from scratch.**

```shell
const MyComponent = () => {
  const validate = (values) => ({
    foo: values > 10,
  });

  return (
    <Block>
      <FormCard initialValues={initialValues} onSubmit={handleSubmit} validate={validate}>
        <FormLayoutRow>
          <FieldNumber id="foo" name="foo" label="Foo" />
        </FormLayoutRow>
      </FormCard>
    </Block>
  );
}
```

## Components

### [FormCard](https://github.com/istreamlabs/pebble-formik/blob/main/src/FormCard.tsx)

> FormCard was designed to be an all-in-one form wrapper, including a prompt message if the user tries to navigate away from the form and a submit and reset button in the footer.

Reviewer 3: ```suggestion
FormCard was designed to be an all-in-one form wrapper with a submit and reset button in the footer. It includes a prompt message if the user tries to navigate away from.
```

* [FieldSelect](https://github.com/istreamlabs/pebble-formik/blob/main/src/FieldSelect.tsx) / [PebbleComponent](https://pebble.istreamplanet.net/#/Components/FieldSelect)
* [FieldText](https://github.com/istreamlabs/pebble-formik/blob/main/src/FieldText.tsx) / [PebbleComponent](https://pebble.istreamplanet.net/#/Components/FieldText)
* [FieldTextDebounce](https://github.com/istreamlabs/pebble-formik/blob/main/src/FieldTextDebounce.tsx) / [PebbleComponent](https://pebble.istreamplanet.net/#/Components/FieldTextDebounce)
* [FieldToggle](https://github.com/istreamlabs/pebble-formik/blob/main/src/FieldToggle.tsx) / [PebbleComponent](https://pebble.istreamplanet.net/#/Components/FieldToggle)

### Error Components

Formats error messages in a standard, easy-to-read manner.

* [ErrorMessage](https://github.com/istreamlabs/pebble-formik/blob/main/src/ErrorMessage.tsx) : Has a `small` prop that fits on a single form row.
* [ErrorSummary](https://github.com/istreamlabs/pebble-formik/blob/main/src/ErrorSummary.tsx) : Is built in to the bottom of FormCard.

> ## How to Release

Reviewer: should be in development file similar to https://github.com/istreamlabs/pebble/blob/main/DEVELOPMENT.md

Author: added https://github.com/istreamlabs/pebble-formik/blob/ux-949/DEVELOPMENT.md

### Release an Update

Package releases should be done through the following commands:

Release a patch update:

```shell

> $ npm version patch -m "release package %s because reasons"

Reviewer: ```suggestion
npm version patch -m "release package %s because reasons"
```

```

Release a minor update:

```shell

> $ npm version minor -m "release package %s because reasons"

Reviewer: ```suggestion
npm version minor -m "release package %s because reasons"
```

```

Alpha pre-release minor update:

```shell

> $ npm version pre-minor -m "release package %s as alpha because reasons" --preid=alpha

Reviewer: ```suggestion
npm version pre-minor -m "release package %s as alpha because reasons" --preid=alpha
```

```

The following will then happen:

* Execute a coverage test run to make sure everything is okay
* Bump the version number in package.json accordingly (see examples)
* Add a tag in git with that version number
* Push to git origin including tags

From there the CI machine takes over and for every tagged commit the following happens:

* NPM publish is run, this in turn will trigger our prepack that cleans and builds the package
* The newly minted package is published with the next tag. This is done to prevent prerelease packages from being installed by default.

### Tag "latest" on npm

To promote the new package to the latest tag and make it the preferred default install:

> 1. Log into with `npm login [username]` but if rumors are true, that could increase a lot in the future.

Reviewer: Rumors do not need to be in here

```suggestion
1. Log into with `npm login [username]`.
```
